sqlc-gen-ftl/src/plugin/mod.rs

```diff
@@ -0,0 +1,187 @@
+#![allow(dead_code)]
+
+use crate::protos::pluginpb;
+use crate::protos::schemapb;
+use crate::protos::schemapb::r#type::Value as TypeValue;
+use prost::Message;
+
+pub struct Plugin;
+
+impl Plugin {
+    pub fn generate_from_input(input: &[u8]) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+        let req = pluginpb::GenerateRequest::decode(input)?;
+        let resp = Self::handle_generate(req)?;
+        Ok(resp.encode_to_vec())
+    }
+
+    fn handle_generate(req: pluginpb::GenerateRequest) -> Result<pluginpb::GenerateResponse, Box<dyn std::error::Error>> {
+        let module = generate_schema(&req)?;
+        Ok(pluginpb::GenerateResponse {
+            files: vec![pluginpb::File {
+                name: "queries.pb".to_string(),
+                contents: module.encode_to_vec(),
+            }],
+        })
+    }
+}
+
+fn generate_schema(request: &pluginpb::GenerateRequest) -> Result<schemapb::Module, Box<dyn std::error::Error>> {
+    let mut decls = Vec::new();
+    let module_name = get_module_name(request)?;
+    
+    for query in &request.queries {
+        if !query.params.is_empty() {
+            decls.push(to_verb_request(query));
+        }
+
+        if !query.columns.is_empty() {
+            decls.push(to_verb_response(query));
+        }
+
+        decls.push(to_verb(query, &module_name));
+    }
+
+    Ok(schemapb::Module {
+        name: module_name,
+        builtin: false,
+        runtime: None,
+        comments: Vec::new(),
+        pos: None,
+        decls,
+    })
+}
+
+fn to_verb(query: &pluginpb::Query, module_name: &str) -> schemapb::Decl {
+    let request_type = if !query.params.is_empty() {
+        Some(to_schema_ref(module_name, &format!("{}Request", query.name)))
+    } else {
+        None
+    };
+
+    let response_type = if query.cmd == ":exec" {
+        None
+    } else {
+        Some(to_schema_ref(module_name, &format!("{}Response", query.name)))
+    };
+
+    schemapb::Decl {
+        value: Some(schemapb::decl::Value::Verb(schemapb::Verb {
+            name: query.name.clone(),
+            export: false,
+            runtime: None,
+            request: request_type,
+            response: response_type,
+            pos: None,
+            comments: Vec::new(),
+            metadata: Vec::new(),
+        })),
+    }
+}
+
+fn to_verb_request(query: &pluginpb::Query) -> schemapb::Decl {
+    schemapb::Decl {
+        value: Some(schemapb::decl::Value::Data(schemapb::Data {
+            name: format!("{}Request", query.name),
```

Comment:
Maybe we should call these "Query" to differentiate them a bit?

Reply:
i like it!